C01/credit-batch-metadata/C01-20150101-20151231-003.json

```diff
@@ -0,0 +1,18 @@
+{
+  "@context": {
+    "schema": "http://schema.org/",
+    "regen": "http://regen.network/"
+  },
+  "@type": "regen:CreditBatch",
+  "regen:vcsRetirementSerialNumber": "5531-243019420-243019439-VCU-048-MER-CD-14-934-01012015-31122015-1",
+  "regen:vcsProjectID": 934,

```

Comment:
it seems like this `regen:vcsProjectID` is not actually part of the json template for vcs credit batches https://github.com/regen-network/regen-registry-standards/blob/main/jsonld/credit-batches/C01-verified-carbon-standard-batch.json, should we update that?

Reply:
Also we're already using `regen:vcsProjectId` in the project metadata: https://github.com/regen-network/regen-registry-standards/blob/main/jsonld/projects/C01-verified-carbon-standard-project.json#L11
we should be consistent with casing and use either ID or Id everywhere

Reply:
@blushi I'm not sure if we need to. The reason we included `regen:vcsProjectID` in the project metadata is because there is no link between batches and projects in our current structure. In the future we intend to include this directly in the project entity via the [`project_key](https://github.com/regen-network/regen-ledger/blob/8231250629db662154b96ae33626524e6efbf598/proto/regen/ecocredit/v1/state.proto#L129-L131). With that in mind I think we maybe don't need to include it in the SHACL schema.

Reply:
> In the future we intend to include this directly in the project entity via the [project_key](https://github.com/regen-network/regen-ledger/blob/8231250629db662154b96ae33626524e6efbf598/proto/regen/ecocredit/v1/state.proto#L129-L131)

did you mean to say the batch entity here @S4mmyb? just wanting to clarify that

i'm also having trouble seeing why we wouldn't want to include the `vcsProjectId/D` in the C01 batch schema, if actual C01 batch metadata includes it.

so what you're saying is that eventually the batch metadata will have no references to projects and that this reference will only live on-chain?

i very well could be misunderstanding but i feel like it's good to have it documented in the C01 schema even if it's only true at this point in time.

Reply:
Ahh good point @wgwz, yes I meant batch entity. And after talking with @ryanchristo about this some more we will actually be using a `reference_id` to tie projects to batches, not the `project_key` which is for internal indexing. The `reference_id` hasn't been implemented quite yet, but there's an open issue for this [here](https://github.com/regen-network/regen-ledger/issues/1066). 

To answer your question we want to include it in the batch schema rather than the batch metadata to support quicker lookups and querying which will be really important in our upcoming bridge use cases for bringing TCO2s over from polygon or from other registries. 

That said, sure we can include the `projectId` in the schema for now and just migrate this in a month or two once we upgrade to 4.0! I'll open a new issue for this in the standards

Reply:
In v3.0 a project has an `id` and `project_id`. In v4.0 a project will have a `key`, `id`, and `reference_id`.

We essentially renamed `id` to `key` and `project_id` to `id` and we are adding a `reference_id`. It was confusing our developers to have an `id` and `project_id` and we assumed it would be confusing to others as well.

The `key` (previously `id`) is used internally for join tables. The `id` (previously `project_id`) is the unique identifier for the project, e.g. `C01-001` (credit class id + project sequence). The `reference_id` can be any arbitrary string with no unique constraints and will be primarily used for convenient lookups when bridging assets.

Reply:
I'm being a bit picky here but there's still this casing issue that hasn't been addressed: `regen:vcsProjectID` vs `regen:vcsProjectId`
but now that the credit batches have been issued on mainnet using `regen:vcsProjectID` and we can't actually update the metadata there, let's just stick to `regen:vcsProjectID` everywhere and update the existing metadata in the off-chain projects where we use `regen:vcsProjectId`

I've created an issue to tackle all these changes: https://github.com/regen-network/regen-registry/issues/923 cc/ @S4mmyb @mhagel @wgwz

Reply:
I've created an issue to add that field on regen-registry-standards https://github.com/regen-network/regen-registry/issues/923